CBinding/Parser/Symbol.cs

```diff
@@ -0,0 +1,204 @@
+			);
+			isDefinition = clang.isCursorDefinition (cursor) != 0;
+			isConst = clang.isConstQualifiedType (clang.getCursorType (cursor)) != 0;
+			spellingLength = Convert.ToInt32 (end.Offset - begin.Offset);
+			spelling = clang.getCursorSpelling (cursor).ToString ();
+			IsDeclaration = clang.isDeclaration (cursor.kind) != 0;
+			IsGlobal = global;
+			Ours = fileN.Equals (begin.FileName);
+		}
+
+		public CProject Project {
+			get {
+				return project;
+			}
+		}
+
+		public string FileName {
+			get {
+				return fileName;
+			}
+		}
+
+		public string Signature {
+			get {
+				return signature;
+			}
+		}
+
+		public string Spelling {
+			get {
+				return spelling;
+			}
+		}
+
+		public string Name {
+			get {
+				return spelling;
+			}
+		}
+
+		public CX_CXXAccessSpecifier Access {
+			get {
+				return access;
+			}
+		}
+
+		public CXCursor Represented {
+			get {
+				return represented;
+			}
+		}
+
+		public CXCursor ParentCursor {
+			get {
+				return parent;
+			}
+		}
+
+		public Symbol Parent {
+			get {
+				try 
+				{
+					switch (parent.kind) {
+					case CXCursorKind.CXCursor_TranslationUnit:
+						return null;
+					case CXCursorKind.CXCursor_Namespace:
+						return new Namespace (project, fileName, parent, IsGlobal);
+					case CXCursorKind.CXCursor_ClassDecl:
+						return new Class (project, fileName, parent, IsGlobal);
+					case CXCursorKind.CXCursor_FieldDecl:
+						return new Field (project, fileName, parent, IsGlobal);
+					case CXCursorKind.CXCursor_ClassTemplate:
+						return new ClassTemplate (project, fileName, parent, IsGlobal);
+					case CXCursorKind.CXCursor_ClassTemplatePartialSpecialization:
+						return new ClassTemplatePartial (project, fileName, parent, IsGlobal);
+					case CXCursorKind.CXCursor_StructDecl:
+						return new Struct (project, fileName, parent, IsGlobal);
+					case CXCursorKind.CXCursor_FunctionDecl:
+						return new Function (project, fileName, parent, IsGlobal);
+					case CXCursorKind.CXCursor_CXXMethod:
+						return new MemberFunction (project, fileName, parent, IsGlobal);
+					case CXCursorKind.CXCursor_FunctionTemplate:
+						return new FunctionTemplate (project, fileName, parent, IsGlobal);
+					case CXCursorKind.CXCursor_EnumDecl:
+						return new Enumeration (project, fileName, parent, IsGlobal);
+					case CXCursorKind.CXCursor_EnumConstantDecl:
+						return new Enumerator (project, fileName, parent, IsGlobal);
+					case CXCursorKind.CXCursor_UnionDecl:
+						return new Union (project, fileName, parent, IsGlobal);
+					case CXCursorKind.CXCursor_TypedefDecl:
+						return new Typedef (project, fileName, parent, IsGlobal);
+					case CXCursorKind.CXCursor_VarDecl:
+						return new Variable (project, fileName, parent, IsGlobal);
+					case CXCursorKind.CXCursor_MacroDefinition:
+						return new Macro (project, fileName, parent, IsGlobal);
+					default:
+						return new Symbol (project, fileName, parent, IsGlobal);
+					}
+				} catch (Exception ex){
```

Comment:
catch-and-ignore is bad. You should at least have a log statement and a comment explaining why.